tools/pre_ci.py

```
@@ -0,0 +1,340 @@
+                sys.stderr.write(e.stderr)
+
+            # Attach captured outputs to the failure record
+            self.record_result(
+                description, False, {"stdout": e.stdout, "stderr": e.stderr}
+            )
+            if fail_fast:
+                sys.exit(e.returncode)
+            return False
+        except (subprocess.SubprocessError, OSError) as e:
+            print(f"\n❌ FATAL: '{description}' generated an exception: {e}")
+            self.record_result(description, False, {"stdout": str(e), "stderr": ""})
```

Ruff (0.15.11)

[warning] Boolean positional value in function call

(FBT003)